src/VisualStudio/Core/Impl/Options/FullSolutionAnalysisOptionBinding.cs

```diff
@@ -0,0 +1,51 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using Microsoft.CodeAnalysis.Options;
+using Microsoft.CodeAnalysis.Shared.Options;
+
+namespace Microsoft.VisualStudio.LanguageServices.Implementation.Options
+{
+    internal class FullSolutionAnalysisOptionBinding
```

Comment:
I believe that is job for caller if they have dependency on FG. this code, it is completely fine, they use it in different code path in BG. it will work just fine.